### PR TITLE
Rename topology module to dynamic.configuration

### LINKED ISSUE #18032: Rename topology module and classes to be more generic
To allow adding more dynamic configurations, we should rename cluster topology module to something more generic.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

We decided to add configurations that can be changed dynamically to the cluster topology and make use of the same infrastructure to apply changes in a systematic way. Hence the name `topology` or `ClusterTopology` is not meaningful anymore. This PR renames most of the references to topology to `configuration` or `ClusterConfiguration`. 
* The package `topology` is renamed to `dynamic.configuration`
* The module is renamed to `dynamic-config`
* All classes have renamed
* Most reference to topology in the comments, variables and method names have been renamed. There might be some still left which we can fix as we go. 

The protobuf messages are not renamed to minimize the impact in this PR. This will be done in a follow up. The gateway rest API responses are also untouched. 

## Related issues

closes #18032

### CODE REVIEW

**Comment by deepthidevaki** [comment]
@npepinpe Sorry, this is a big PR :see_no_evil: But I guess you don't have to go through every change. Please check mainly the renaming for package and module. 

Also realized that it will be a pain to backport any bug fixes after this :sob:

File: dist/src/main/java/io/camunda/zeebe/gateway/TopologyServices.java
Comment (npepinpe) [inline dist/src/main/java/io/camunda/zeebe/gateway/TopologyServices.java:1]: :upside_down_face: Would rename to `ClusterConfigServices` or `DynamicConfigServices` or something. Same in the broker.

**Review by npepinpe — APPROVED** [review]
:thought_balloon: `dynamic.configuration` is quite long :smile: I would've just named it `cluster.config` or something :shrug: After all, most classes after are just called `ClusterConfiguration` something. Plus, the artifact is called `zeebe-cluster-configuration` (could be shortened to `zeebe-cluster-config`). Not a big deal though, and I can't articulate a better argument, so it's fine.

:thought_balloon: I think we may want a topology listener interface in the future for consumers who only care about topology updates, and not more general config, but it's still too early to say.

**Comment by deepthidevaki** [comment]
> 💭 `dynamic.configuration` is quite long 😄 I would've just named it `cluster.config` or something 🤷 After all, most classes after are just called `ClusterConfiguration` something. Plus, the artifact is called `zeebe-cluster-configuration` (could be shortened to `zeebe-cluster-config`). Not a big deal though, and I can't articulate a better argument, so it's fine.

Done. Renamed to `dynamic.config`.

> 
> 💭 I think we may want a topology listener interface in the future for consumers who only care about topology updates, and not more general config, but it's still too early to say.

It might be useful. Let's add it when it is needed.

**Comment by deepthidevaki** [comment]
Rebasing on main.

**Pull request merged**